src/main/resources/META-INF/plugin.xml

```diff
@@ -30,11 +30,10 @@
   <depends>com.intellij.modules.platform</depends>
 
   <application-components>
-    <component>
-      <implementation-class>com.github.gilday.darkmode.DarkModeSync</implementation-class>
-    </component>
   </application-components>
   <extensions defaultExtensionNs="com.intellij">
+    <postStartupActivity implementation="com.github.gilday.darkmode.DarkModeSync$MyStartupActivity"/>
```

Comment:
`postStartupActivity` pertains to the Project lifecycle. Can we use an [AppLifecycleListener](https://upsource.jetbrains.com/idea-ce/file/idea-ce-f0b3942b49880400994f62c064afd4ef909539eb/platform/platform-impl/src/com/intellij/ide/AppLifecycleListener.java) instead? I'm thinking we may need to initialize the singleton in both the `welcomeScreenDisplayed` and `appStarting` callbacks. Thoughts?

Reply:
I've migrated `MyStartupActivity` into a `MyLifecycleListener` which implements `AppLifecycleListener`. 

Also `DarkModeSync` no longer implements `Disposable`. Instead it is all handled in the following lifecycle methods.
1. `appStarting()` -> calls `onStartup()`
2. `appWillBeClosed()` -> `calls onClose()`

And there's no longer a need to pass a project to `getInstance()`, which just returns a singleton.